charmhelpers/core/hookenv.py

```diff
@@ -967,6 +967,8 @@ def charm_dir():
 @cached
 def action_get(key=None):
     """Gets the value of an action parameter, or all key/value param pairs"""
+    log("action_get is deprecated. Please consider using function_get "
+        "instead.", level=WARNING)
```

Comment:
There is a handy `deprecate` decorator in the `charmhelpers/__init__.py` which can be used; I'm not sure how it plays with cached, so it should probably go after the `@cached` decorator to maximise caching.

Reply:
Thanks! Fixed.